docs/reference/connector/apis/set-connector-sync-job-stats-api.asciidoc

```diff
@@ -0,0 +1,74 @@
+[[set-connector-sync-job-stats-api]]
+=== Set connector sync job stats API
+++++
+<titleabbrev>Set connector sync job stats</titleabbrev>
+++++
+
+Sets connector sync job stats.
+
+[[set-connector-sync-job-stats-api-request]]
+==== {api-request-title}
+`PUT _connector/_sync_job/<connector_sync_job_id>/_stats`
+
+[[set-connector-sync-job-stats-api-prereqs]]
+==== {api-prereq-title}
+
+* To sync data using connectors, it's essential to have the Elastic connectors service running.
+* The `connector_sync_job_id` parameter should reference an existing connector sync job.
+
+[[set-connector-sync-job-stats-api-desc]]
+==== {api-description-title}
+
+Sets the stats for a connector sync job.
+Stats include: `deleted_document_count`, `indexed_document_count`, `indexed_document_volume` and `total_document_count`.
+`last_seen` can also be update using this API.
+This API is mainly used by the connector service for updating sync job information.
+
+[[set-connector-sync-job-stats-api-path-params]]
+==== {api-path-parms-title}
+
+`<connector_sync_job_id>`::
+(Required, string)
+
+[role="child_attributes"]
+[[set-connector-sync-job-stats-api-request-body]]
+==== {api-request-body-title}
+
+`deleted_document_count`::
+(Required, int) The document count the sync job deleted.
+
+`indexed_document_count`::
+(Required, int) The document count the sync job indexed.
+
+`indexed_document_volume`::
+(Required, int) The document volume the sync job indexed.
+
+`total_document_count`::
+(Optional, int) The total document count in the target index after the sync job finished.
+
+`last_seen`::
+(Optional, instant) The timestamp to set `last_seen` of the connector sync job to.
+
+[[set-connector-sync-job-stats-api-response-codes]]
+==== {api-response-codes-title}
```

Comment:
Guess we should have a 404 too right?

Reply:
Yes good catch, added it with https://github.com/elastic/elasticsearch/pull/103469/commits/9a489f92a1a977c40d1a24148ee0f7624de4d30a